praw/models/listing/generator.py

```diff
@@ -70,7 +71,7 @@ def __next__(self) -> Any:
         self.yielded += 1
         return self._listing[self._list_index - 1]
 
-    def _extract_sublist(self, listing: dict[str, Any] | list[Any]):
+    def _extract_sublist(self, listing: dict[str, Any] | list[Any] | Any) -> Any:
```

Comment:
Doesn't the `Any` in the parameters signature nullify the rest of the annotations?

Reply:
I believe I _only_ needed to add the return type to satisfy the check. However, while I was looking at it this it seemed like we might be accepting other types than just dict and list. I could be mistaken about that.

Reply:
I'd have to look in my IDE to know for sure.

Reply:
I made some fixes to resolve pyright issues in generator.py and listing.py.